src/spyder/der.py

```diff
@@ -4,15 +4,23 @@
 from tabulate import tabulate
 
 from collections import defaultdict
+import numpy as np
 
 
 class DERMetrics:
     def __init__(self, metrics):
-        self.duration = metrics.duration
-        self.miss = metrics.miss
-        self.falarm = metrics.falarm
-        self.conf = metrics.conf
-        self.der = metrics.der
+        try:
+            self.duration = metrics.duration
+            self.miss = metrics.miss
+            self.falarm = metrics.falarm
+            self.conf = metrics.conf
+            self.der = metrics.der
+        except:
+            self.duration = metrics["duration"]
```

Comment:
Why is this needed? Also, you should specify the exception type (bare except is not recommended as per pep8 rules).

Reply:
Because I used `dict` in line 90:
```
    return {m[0]: DERMetrics(dict(zip(['duration', 'miss', 'falarm', 'conf', 'der'], m[1:])))
            for m in selected_metrics}
```

Would you mind if I force-pushed this part as follows?
```
        if isinstance(metrics, dict):
            self.duration = metrics["duration"]
            self.miss = metrics["miss"]
            self.falarm = metrics["falarm"]
            self.conf = metrics["conf"]
            self.der = metrics["der"]
        else:
            self.duration = metrics.duration
            self.miss = metrics.miss
            self.falarm = metrics.falarm
            self.conf = metrics.conf
            self.der = metrics.der
```

Reply:
Is it necessary to return it as a dict? Could you just return it as an object of type `DERMetrics`?

Reply:
I've pushed a new commit. Thanks for reviewing.